plugins/exporter.koplugin/main.lua

```diff
@@ -1,431 +1,70 @@
-        if type(settings) == "table" then
-            settings.domain = nil
-            settings.username = nil
-            settings.token = nil
+    end
+
+    if type(settings) == "table" then
+        for _, fmt in ipairs(formats) do
+            if type(settings[fmt]) == "table" then return end
         end
+        local new_settings = {}
+        for _, fmt in ipairs(formats) do
+            new_settings[fmt] = { enabled = false }
+        end
+        new_settings["joplin"].ip = settings.joplin_IP
+        new_settings["joplin"].port = settings.joplin_port
+        new_settings["joplin"].token = settings.joplin_token
+        new_settings["readwise"].token = settings.readwise_token
+        G_reader_settings:saveSetting("exporter", new_settings)
     end
-    return settings or {}
 end
 
-local Exporter = InputContainer:new{
+local Exporter = InputContainer:new {
     name = "exporter",
-    notebook_name = _("KOReader Notes"),
-    notemarks = _("Note: "),
     clipping_dir = DataStorage:getDataDir() .. "/clipboard",
-
-    evernote_token = nil,
-    notebook_guid = nil,
+    targets = {
+        html = require("formats/html"),
+        joplin = require("formats/joplin"),
+        json = require("formats/json"),
+        readwise = require("formats/readwise"),
+        text = require("formats/text"),
+    },
 }
 
 function Exporter:init()
-    self.text_clipping_file = self.clipping_dir .. "/KOReaderClipping.txt"
-    self.json_clipping_file = self.clipping_dir .. "/KOReaderClipping.json"
-    local settings = getOrMigrateSettings()
-    self.notebook_guid = settings.notebook
-    self.joplin_IP = settings.joplin_IP or "localhost"
-    self.joplin_port = settings.joplin_port or 41185
-    self.joplin_token = settings.joplin_token -- or your token
-    self.joplin_notebook_guid = settings.joplin_notebook_guid or nil
-    self.readwise_token = settings.readwise_token or nil
-    self.html_export = settings.html_export or false
-    self.joplin_export = settings.joplin_export or false
-    self.txt_export = settings.txt_export or false
-    self.json_export = settings.json_export or false
-    self.readwise_export = settings.readwise_export or false
-    --- @todo Is this if block necessary? Nowhere in the code they are assigned both true.
-    -- Do they check against external modifications to settings file?
-
-    if self.html_export then
-        self.txt_export = false
-        self.joplin_export = false
-        self.json_export = false
-        self.readwise_export = false
-    elseif self.txt_export then
-        self.joplin_export = false
-        self.json_export = false
-        self.readwise_export = false
-    elseif self.json_export then
-        self.joplin_export = false
-        self.readwise_export = false
-    end
-
-    self.parser = MyClipping:new{
+    migrateSettings()
+    self.parser = MyClipping:new {
         my_clippings = "/mnt/us/documents/My Clippings.txt",
         history_dir = "./history",
```

Comment:
This got me. I'm not sure how it works on android.

Where rootdir is `/data/data/org.koreader.launcher/files` but we keep user settings in `/sdcard/koreader`. 

I *think* history is saved in `dataDir` (`/sdcard/koreader`) for all documents where is not possible to store highlights in sidecar (because the filesystem lacks writable permission). I'm going to double check this